app/views/topology/service.js

```diff
@@ -1346,6 +1349,29 @@ YUI.add('juju-topology-service', function(Y) {
     },
 
     /**
+      Returns a service name which is a maximum of 10 characters wide.
+
+      @method truncateServiceName
+      @param {Object} service The service model
+      @return {String} The truncated service display name.
+    */
+    truncateServiceName: function(service) {
+      var name = service.displayName;
```

Comment:
yea, I think this would be nicer as either a method on the service so you can just do d.truncateServiceName which would then use self.displayName and self.pending or that it was a helper that took the two properties name, pending as arguments.